### PR TITLE
Browser support

### PR DESCRIPTION
**Pull request by jamesramsay**
Requires a custom `resolveLink` function to be provided which is responsible for handling link inflation.

### CODE REVIEW

**Comment by jamesramsay** [comment]
@jmatsushita please could you take a look at this PR and see if it fits the bill?

**Comment by Ramblurr** [comment]
Could you document the `resolveLink` option in the README?

**Comment by Ramblurr** [comment]
Also, for non-browser implementations, it would be great to be able to rely on the default implementation of `linkResolver`.

My use case is where I want to be able to handle some custom links, but also support the base feature of including files via paths.

**Comment by jamesramsay** [comment]
@Ramblurr I'll absolutely add documentation before the branch merges, just haven't had a chance yet.

In this first version I hadn't really planned on supporting extending the base implementation, but perhaps I should reevaluate.

**Comment by Ramblurr** [comment]
@jamesramsay Cool! 

Another piece of feedback, it doesn't seem to work with nested transcludes.

I wrote a simple `customLinkResolver` that calls the default implementation in `resolve.js`, but my version is only passed the transcludes in the base file. The nested transcludes are never passed to the custom resolver.

Is this a bug or intended?

**Comment by jamesramsay** [comment]
@Ramblurr that's a bug. I'll be sure to add a test for that 😉

**Comment by Ramblurr** [comment]
Awesome! I am stoked for this feature, it will really allow for some great opportunities (Such as #270)

Another possibility is some short-hand notation for api blueprint projects.

For example we include our JSON body's with a transclude  `:[a task](../../models/task.json)`, but it is cumbersome to remember the number of `../` you need. With the custom link resolver, we can shorten this to simply `:[a task](models/task)` 

But yea, having nested transcludes working is a must.

BTW, I simply added `export { resolveLink } from './resolve'`  to hercule.js to expose the default resolveLink implementation to clients.

**Comment by Ramblurr** [comment]
Any updates on this issue? We're using our [own forked hercule](https://github.com/Ramblurr/hercule/commit/95a6dfaece2e8961248b7cd5fbe28a80c33ab475) right now because we need access to the link resolver. Note that we aren't even using hercule in the browser, we just wanted to write our own extensions to the transclude syntax to support shortcuts and other processing.

**Comment by jamesramsay** [comment]
@Ramblurr sorry for the lack of updates. It had really fallen off my radar. I'll try to dust this off and publish an MVP without browser considerations in the near future.

**Comment by jamesramsay** [comment]
@Ramblurr I split out the portion not related to browser support and released it in v3.1.0. I haven't added documentation yet because I want to clean up the API make it easier to use first.

**Comment by jamesramsay** [comment]
@Ramblurr I realised I never left a comment to let you know hercule 4.x has a documented API for custom resolvers. Feel free to open an issue if you have any questions!

**Comment by jamesramsay** [comment]
Closing this PR due to age. Latest release could be used in browser with a custom resolve.
